### PR TITLE
arvdias/add new metrics

### PR DESCRIPTION
**Pull request by arvdias**

### CODE REVIEW

File: src/metric/populate_test.go [hunk 1]
```diff
@@ -147,13 +147,18 @@ func TestPopulateK8s(t *testing.T) {
...
+	// TODO not good to compare error strings...
 	expectedErrs := []error{
 		errors.New("error populating metric for entity ID kube-system_newrelic-infra-rz225: cannot fetch value for metric deploymentName, metric not found"),
 		errors.New("error populating metric for entity ID kube-system_newrelic-infra-rz225: cannot fetch value for metric reason, metric not found"),
 		errors.New("error populating metric for entity ID kube-system_newrelic-infra-rz225: cannot fetch value for metric message, metric not found"),
 		errors.New("error populating metric for entity ID kube-system_newrelic-infra-rz225_newrelic-infra: cannot fetch value for metric deploymentName, metric not found"),
 		errors.New("error populating metric for entity ID kube-system_newrelic-infra-rz225_newrelic-infra: cannot fetch value for metric cpuLimitCores, metric not found"),
 		errors.New("error populating metric for entity ID kube-system_newrelic-infra-rz225_newrelic-infra: cannot fetch value for metric reason, metric not found"),
+		errors.New("error populating metric for entity ID kube-system_newrelic-infra-rz225_newrelic-infra: cannot fetch value for metric cpuCoresUtilization, 'cpuUsedCores' is nil"),
```
Comment (paologallinaharbur) [inline src/metric/populate_test.go:158]: Should we at least create a constant for these expected values?

Comment (paologallinaharbur) [reply src/metric/populate_test.go:158]: I guess it is not ideal since they are kind of generated 😕

Comment (arvdias) [reply src/metric/populate_test.go:158]: yes, checking strings is not the best option. We should create some kind of error type for these. I just didn't have time in this pr

File: src/metric/definition.go [hunk 2]
```diff
@@ -974,3 +1068,31 @@ func toCores(value definition.FetchedValue) (definition.FetchedValue, error) {
...
+		return float64(v), nil
+	case prometheus.CounterValue:
+		return float64(v), nil
+	}
+
+	return nil, fmt.Errorf("invalid type value '%v'. Expected 'gauge' or 'counter', got '%T'", value, value)
+}
+
+// Subtract returns a new FetchFunc that subtracts 2 values. It expects that the values are float64
+func Subtract(left definition.FetchFunc, right definition.FetchFunc) definition.FetchFunc {
```
Comment (paologallinaharbur) [inline src/metric/definition.go:1084]: Now we understand when you said finding where to add these function was not straightforward 😅

File: src/metric/definition.go [hunk 3]
```diff
@@ -405,6 +405,12 @@ var EtcdSpecs = definition.SpecGroups{
 				ValueFunc: prometheus.FromValueWithOverriddenName("go_goroutines", "goGoroutines"),
 				Type:      sdkMetric.GAUGE,
 			},
+			// computed
```
Comment (paologallinaharbur) [inline src/metric/definition.go:408]: is it any difference between `computed` and `calculated` metrics?

File: src/metric/definition.go (hunk 2 above)
Comment (arvdias) [reply src/metric/definition.go:1084]: yeah, it's all a bit complex

File: src/metric/definition.go (hunk 3 above)
Comment (arvdias) [reply src/metric/definition.go:408]: no, it's probably I just that I did it at different times

File: src/metric/populate_test.go (hunk 1 above)
Comment (arvdias) [reply src/metric/populate_test.go:158]: Let's do a release a first and then  I'll do another PR to change this. It's not a critical issue at the moment

File: src/metric/definition.go [hunk 4]
```diff
@@ -495,6 +506,11 @@ var KSMSpecs = definition.SpecGroups{
 			{Name: "statefulsetName", ValueFunc: prometheus.FromLabelValue("kube_statefulset_created", "statefulset"), Type: sdkMetric.ATTRIBUTE},
 			{Name: "namespaceName", ValueFunc: prometheus.FromLabelValue("kube_statefulset_created", "namespace"), Type: sdkMetric.ATTRIBUTE},
 			{Name: "label.*", ValueFunc: prometheus.InheritAllLabelsFrom("statefulset", "kube_statefulset_labels"), Type: sdkMetric.ATTRIBUTE},
+			//computed
+			{Name: "podsMissing", ValueFunc: Subtract(
```
Comment (paologallinaharbur) [inline src/metric/definition.go:510]: Is it normal that the name is always podsMissing?

Comment (paologallinaharbur) [reply src/metric/definition.go:510]: Is it the same attribute saved in different sample?

Comment (paologallinaharbur) [reply src/metric/definition.go:510]: Yes, I believe it is that, than it makes sense

**Review by paologallinaharbur — APPROVED** [review]

**Comment by arvdias** [comment]
closes #56

**Pull request merged**